File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    let obj = {};
    
    for(let i=0; i<tangerine.length; i++){
        obj[tangerine[i]] ? obj[tangerine[i]]++ : obj[tangerine[i]] = 1;
    }
    
    let arr = Object.values(obj).sort((a,b)=> b-a);
    
    for(let j of arr){
        k -= j;
        answer++;
        if(k <=0 ) break;
    }
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;

        HashMap<Integer, Integer> map = new HashMap<>();

        for (int size : tangerine) {
            map.put(size, map.getOrDefault(size, 0) + 1);
        }

        ArrayList<Integer> countList = new ArrayList<>(map.values());

        countList.sort(Collections.reverseOrder());

        for (int count : countList) {
            k -= count;
            answer++;

            if (k <= 0) {
                break;
            }
        }

        return answer;
    }
}